.github/workflows/ci.yml

```diff
@@ -0,0 +1,63 @@
+name: CI
+
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build:
+  
+    strategy:
+      matrix:
+        target:
+          - linux
+          - windows
+          - macos
+        include:
+          - target: linux
+            os: ubuntu-latest
+            artifact: endless-sky-editor
+          - target: windows
+            os: windows-latest
+            artifact: release/endless-sky-editor.exe
+          - target: macos
+            os: macos-latest
+            artifact: endless-sky-editor.app/
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+
+    - uses: actions/checkout@v2
+
+    - name: Cache Qt
+      id: cache-qt
+      uses: actions/cache@v1
+      with:
+        path: ${{ github.workspace }}/Qt/
+        key: ${{ runner.os }}-QtCache
+
+    - name: Install Qt
+      uses: jurplel/install-qt-action@v2
+      with:
+        version: "5.15.2"
+        dir: ${{ github.workspace }}/Qt/
+        arch: win64_mingw81 # This is only used (and needed) on windows, so we can safely specify it for all OSes
+        cached: ${{ steps.cache-qt.outputs.cache-hit }}
+        
+    - name: Run qmake
+      run: qmake
+      
+    - name: Run make
+      run: make -j $(nproc)
+
+    - name: Upload artifact
```

Comment:
Can do, but it's a mediocre check at best, because on this machine all dependencies are installed. We could move the check into a separate step, so it runs on a clean machine.

Reply:
Both are fine. It's certainly not a robust check, but does ensure we didn't just build garbage.